lightning/src/ln/channelmanager.rs

```diff
@@ -3408,7 +3408,13 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					}
 					return Err(MsgHandleErrInternal::send_err_msg_no_close("Got a channel_update for a channel from the wrong node - it shouldn't know about our private channels!".to_owned(), chan_id));
 				}
-				try_chan_entry!(self, chan.get_mut().channel_update(&msg), channel_state, chan);
+				let were_node_one = self.get_our_node_id().serialize()[..] < chan.get().get_counterparty_node_id().serialize()[..];
+				let msg_from_node_one = msg.contents.flags & 1 == 0;
```

Comment:
Might be nicer to make `ChannelFlags` into its own struct and have methods to check who's node_one, etc. Not necessary for this PR though

Reply:
Yea, that's probably a good idea for the `channel_state` in `Channel` too, but we can do it later.